projects/packages/forms/src/blocks/shared/settings/transforms.js

```diff
@@ -129,6 +129,22 @@ const createTextFieldInnerBlocks = ( blockName, existingInnerBlocks = [] ) => {
 
 	const existingLabel = existingInnerBlocks.find( block => block.name === 'jetpack/label' );
 	const existingInput = existingInnerBlocks.find( block => block.name === 'jetpack/input' );
+	const existingInputAttributes = existingInput?.attributes ?? {};
+
+	if ( blockName === 'jetpack/field-select' ) {
+		/**
+		 * When transforming to a select field, add a default placeholder to the input block if none exists.
+		 * The select block's edit.js checks for the placeholder attribute on the input block,
+		 * and add the `has-placeholder` class to the field if it exists.
+		 * However if the incoming transformed block doesn't have a placeholder, the edit.js will
+		 * not add the class because it's trying to honor the placeholder attribute,
+		 * BUT adds a default placeholder anyway.
+		 */
+
+		if ( ! existingInputAttributes?.placeholder ) {
+			existingInputAttributes.placeholder = __( 'Select one option', 'jetpack-forms' );
+		}
+	}
```

Comment:
I've reinstated this change. Seems to be the cleanest spot for it. 

As the comment suggests, the `has-placeholder` class name is required when there's a placeholder. 

However, during a transform, if a placeholder doesn't exist the edit.js will honor this choice and skip the class name.

BUT the select default template kicks in and adds the placeholder anyway. 🤷🏻